src/sas/sascalc/fit/BumpsFitting.py

```diff
@@ -341,21 +338,13 @@ def fit(self, msg_q=None,
                             isinstance(p.value, uncertainties.core.AffineScalarFunc):
                         # value.n returns value p
                         pvec.append(p.value.n)
-                        # value.n returns error in p
+                        # value.s returns error in p
                         stderr.append(p.value.s)
                     # p constrained based on another parameter
+                    # Warning: This will not work for parameters that are tied
+                    # to other parameter already constrained.
+                    # Is this still needed with the added _allComputedParamsUncertaintiesDefined test?
                     else:
```

Comment:
Do you ever get into the `else` condition?

I think you can replace the entire if-else block with:
```
assert isinstance(p.value, (uncertainties.core.Variable, uncertainties.core.AffineScalarFunc))
# value.n returns value p
pvec.append(p.value.n)
# value.s returns error in p
stderr.append(p.value.s)
```

Reply:
Now that everything seems fixed, the execution flow is not entering the else condition any more in any of my tests. So I removed the else block entirely. But as a measure of precaution, I kept a try/except, so that if by any strange reason a parameter is still not an uncertainty object, at least the user will get the parameter value (+ some error message). But if this is judge completely unnecessary, I do not have any objection in removing the try/except.

Reply:
My inclination would be to drop it, but I don't object to leaving it in.

In principle one should strive for 100% code coverage with tests. The except block is untestable (you can't trigger the except without extraordinary measures) . Let the higher up exception handler catch future broken code.

Reply:
Let's do that then. I just removed a few more lines :-)